source/extensions/filters/http/cache/cache_headers_utils.h

```diff
@@ -98,6 +103,42 @@ class CacheHeadersUtils {
    */
   static absl::optional<uint64_t> readAndRemoveLeadingDigits(absl::string_view& str);
 };
+
+class VaryHeader {
+public:
+  // Checks if the headers contain an allowed value in the Vary header.
+  static bool isAllowed(const absl::flat_hash_set<std::string>& allowed_headers,
+                        const Http::ResponseHeaderMap& headers);
+
+  // Checks if the headers contain a non-empty value in the Vary header.
+  static bool hasVary(const Http::ResponseHeaderMap& headers);
+
+  // Creates a single string combining the values of the varied headers from entry_headers.
+  static std::string createVaryKey(const Http::HeaderEntry* vary_header,
+                                   const Http::RequestHeaderMapPtr& entry_headers);
+
+  // Parses the header names that are in the Vary header value.
+  static std::vector<std::string> parseHeaderValue(const Http::HeaderEntry* vary_header);
+
+  // Returns a header map containing the subset of the original headers that can be varied from the
+  // request.
+  static Http::RequestHeaderMapPtr
+  possibleVariedHeaders(const absl::flat_hash_set<std::string>& allowed_headers,
+                        const Http::RequestHeaderMap& request_headers);
+
+  // Parses the allowlist of header values that can be used to create varied responses.
+  static absl::flat_hash_set<std::string> parseAllowlist();
+
+private:
+  // The separator characters are used to create the vary-key, and must be characters that are
+  // invalid to be inside values and header names. The chosen characters are invalid per:
+  // https://tools.ietf.org/html/rfc2616#section-4.2.
+  // Used to separate the values of different headers.
+  inline static const std::string header_separator = "\n";
```

Comment:
The purpose of a header is to share code among multiple source files. header_separator and in_value_separator are only used in one source file, so there's no reason to declare them here. Declare them in an unnamed namespace in the source file.